introduce 'log_connection_messages' GUC setting

sdudoladov opened the pull request:
this patch removes 'log_connections' and 'log_disconnections' in favor
of 'log_connection_messages', thereby introducing incompatibility

Author: Sergey Dudoladov

Reviewed-by:

Discussion:
https://www.postgresql.org/message-id/flat/CAA8Fd-qCB96uwfgMKrzfNs32mqqysi53yZFNVaRNJ6xDthZEgA%40mail.gmail.com

hughcapet commented on src/backend/commands/variable.c:
```diff
@@ -33,6 +33,8 @@
 #include "utils/timestamp.h"
 #include "utils/varlena.h"
 
+#include "postmaster/postmaster.h"
```
why does this include stand apart from the others?

sdudoladov replied:
leftover from an earlier version

hughcapet commented on src/backend/commands/variable.c:
```diff
@@ -933,3 +935,97 @@ show_role(void)
+		return true;
+	}
+
+	if (strcmp(*newval, "all") == 0)
```
```suggestion
	if (pg_strcasecmp(*newval, "all") == 0)
```
Case-independent is better here and further, I guess

sdudoladov replied:
I see both `pg_strcasecmp` and `strcmp` are used [src/backend/commands/variable.c](https://github.com/sdudoladov/postgres/pull/7/files/471f295df8a473d92e9389357f37d6d166866790#diff-9bee4855e89439f420d8845f9c5ff4a9df8da1e186c0a6b1e3de24d2f17869a4).

In that particular case `SplitIdentifierString` actually does the downcasing but I've swtiched to `pg_strcasecmp` anyway

hughcapet commented on src/backend/commands/variable.c:
```diff
@@ -33,6 +33,8 @@
 #include "utils/timestamp.h"
 #include "utils/varlena.h"
 
+#include "postmaster/postmaster.h"
```
Why does this include stay apart?

hughcapet replied:
and what is it required for?

sdudoladov replied:
leftover from an earlier patch version, removed

hughcapet commented on src/backend/commands/variable.c:
```diff
@@ -933,3 +935,97 @@ show_role(void)
+
+	*extra = (void *) myextra;
+
+	if (log_connection_messages == NULL || log_connection_messages[0] == '\0') {
```
`SplitIdentifierString` already considers empty strings as ok, can't we get rid of this string?

sdudoladov replied:
I've decided to keep it for now for the sake of being explicit in the code

hughcapet commented on src/include/postgres.h:
```diff
@@ -805,4 +805,10 @@ extern Datum Float8GetDatum(float8 X);
 #define Float8GetDatumFast(X) PointerGetDatum(&(X))
 #endif
 
+extern PGDLLIMPORT char *log_connection_messages_string;
```
Could you please explain, why do we need to define `extern PGDLLIMPORT` for `log_connection_messages_string`? Is it used anywhere else apart from guc.c?

sdudoladov replied:
true, not used anywhere excep guc.c I've dropped `extern PGDLLIMPORT`

hughcapet commented:
\+ tests should be extended to check the new GUC values, I guess

sdudoladov commented:
> + tests should be extended to check the new GUC values, I guess

yes, but let's wait with that untl the 1st round of feedback from the community

sdudoladov commented:
[relevant refactoring](https://github.com/postgres/postgres/commit/0a20ff54f5e66158930d5328f89f087d4e9ab400)

Commit:
Introduce 'log_connection_messages'

This patch removes 'log_connections' and 'log_disconnections'
in favor of 'log_connection_messages', thereby introducing incompatibility

Author: Sergey Dudoladov

Reviewed-by: Justin Pryzby

Discussion:
https://www.postgresql.org/message-id/flat/CAA8Fd-qCB96uwfgMKrzfNs32mqqysi53yZFNVaRNJ6xDthZEgA%40mail.gmail.com

hughcapet commented on src/backend/commands/variable.c:
```diff
@@ -993,6 +993,74 @@ show_role(void)
+	if (pg_strcasecmp(*newval, "all") == 0)
+	{
+		newlogconnect |= LOG_CONNECTION_ALL;
+		myextra = (int *) guc_malloc(ERROR, sizeof(int));
```
shouldn't we check if `myextra` is NULL

sdudoladov replied:
[we don't need it](https://github.com/postgres/postgres/blob/e4e89eb5bbfdae30349b38344e9c604411174f6b/src/backend/utils/misc/guc.c#L622-L641)

hughcapet commented:
what's about https://github.com/sdudoladov/postgres/blob/make-logging-connections-fine-grained/src/backend/utils/misc/guc_tables.c#L90?
is it still needed?

sdudoladov commented:
> what's about https://github.com/sdudoladov/postgres/blob/make-logging-connections-fine-grained/src/backend/utils/misc/guc_tables.c#L90?
> is it still needed?

obviously not. Thank you !
